umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso19115_2/spatial.clj

```diff
@@ -114,14 +115,190 @@
+                xdimension (when xdimension-index
+                             (get-substring-with-index resolution-string indexes xdimension-index))
+                ydimension (when ydimension-index
+                             (get-substring-with-index resolution-string indexes ydimension-index))
+                unit (when unit-index
+                       (get-substring-with-index resolution-string indexes unit-index))
+                minimum-ydimension (when minimum-ydimension-index
+                                     (get-substring-with-index resolution-string indexes minimum-ydimension-index))
+                maximum-ydimension (when maximum-ydimension-index
+                                     (get-substring-with-index resolution-string indexes maximum-ydimension-index))
+                minimum-xdimension (when minimum-xdimension-index
+                                     (get-substring-with-index resolution-string indexes minimum-xdimension-index))
+                maximum-xdimension (when maximum-xdimension-index
+                                     (get-substring-with-index resolution-string indexes maximum-xdimension-index))
+                scan-direction (when scan-direction-index
+                                 (get-substring-with-index resolution-string indexes scan-direction-index))
+                viewing-angle-type (when viewing-angle-type-index
+                                     (get-substring-with-index resolution-string indexes viewing-angle-type-index))
+                horizontal-resolution-processing-level-enum (when horizontal-resolution-processing-level-enum-index
+                                                              (get-substring-with-index resolution-string indexes horizontal-resolution-processing-level-enum-index))]]
+      (util/remove-nil-keys
+       {:XDimension (when (seq xdimension)
+                      (read-string xdimension))
+        :YDimension (when (seq ydimension)
+                      (read-string ydimension))
+        :Unit (when (seq unit)
+                unit)
+        :MinimumYDimension (when (seq minimum-ydimension)
+                             (read-string minimum-ydimension))
+        :MaximumYDimension (when (seq maximum-ydimension)
+                             (read-string maximum-ydimension))
+        :MinimumXDimension (when (seq minimum-xdimension)
+                             (read-string minimum-xdimension))
+        :MaximumXDimension (when (seq maximum-xdimension)
+                             (read-string maximum-xdimension))
+        :ScanDirection (when (seq scan-direction)
+                         scan-direction)
+        :ViewingAngleType (when (seq viewing-angle-type)
+                            viewing-angle-type)
+        :HorizontalResolutionProcessingLevelEnum (when (seq horizontal-resolution-processing-level-enum)
+                                                   horizontal-resolution-processing-level-enum)}))))
```

Comment:
Nitpick: is there a way to split the `let` above into a few smaller pieces? Some functions like `get-x-dimension`, `get-y-dimension`, etc might make this more readable.